proto/viam/app/data/v1/data.proto

```diff
@@ -123,3 +116,13 @@ message BinaryDataByIDsResponse {
   repeated BinaryData data = 2;
   int64 count = 3;
 }
+
+message BinaryMetadata {
+  string id = 1;
```

Comment:
Should remove id here and just keep in BinaryData to represent the file ID.

Reply:
I put it here so it would be in the resulting json. We already store it in the file name, but I figured this would allow the json to still be valid even with moves/renames, and would allow all metadata to be parsed just from the json (vs most of it living in the json, and the id only existing in the local file name)

Reply:
Based on our offline chat, I would either change the index in BinaryData to `metadata_index` for consistency with tabular, or remove it since it'll always be the index in metadata.

Reply:
I went ahead and just moved metadata to be a part of BinaryData so we don't have to deal with indexes now that they're 1:1, they're just stored in the same message

Reply:
Lmk if you think that makes sense

Reply:
Chatted offline, agree this makes sense. Can rename TabularData later to make it clear that it's not equivalent to BinaryData, but don't want to overload that in today's changes